health/health.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package health
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vechain/thor/v2/thor"
+)
+
+type BlockIngestion struct {
+	ID        *thor.Bytes32 `json:"id"`
+	Timestamp *time.Time    `json:"timestamp"`
+}
+
+type Status struct {
+	Healthy           bool            `json:"healthy"`
+	BlockIngestion    *BlockIngestion `json:"blockIngestion"`
+	ChainBootstrapped bool            `json:"chainBootstrapped"`
+}
+
+type Health struct {
+	lock              sync.RWMutex
+	newBestBlock      time.Time
+	bestBlockID       *thor.Bytes32
+	bootstrapStatus   bool
+	timeBetweenBlocks time.Duration
+}
+
+const delayBuffer = 5 * time.Second
+
+func NewSolo(timeBetweenBlocks time.Duration) *Health {
+	return &Health{
+		timeBetweenBlocks: timeBetweenBlocks + delayBuffer,
+		// there is no bootstrap in solo mode
+		bootstrapStatus: true,
+	}
+}
+func New(timeBetweenBlocks time.Duration) *Health {
+	return &Health{
+		timeBetweenBlocks: timeBetweenBlocks + delayBuffer,
+	}
+}
+
+func (h *Health) Status() (*Status, error) {
+	h.lock.RLock()
+	defer h.lock.RUnlock()
+
+	blockIngest := &BlockIngestion{
+		ID:        h.bestBlockID,
+		Timestamp: &h.newBestBlock,
+	}
+
+	healthy := time.Since(h.newBestBlock) <= h.timeBetweenBlocks && // less than 10 secs have passed since a new block was received
+		h.bootstrapStatus
+
+	return &Status{
+		Healthy:           healthy,
+		BlockIngestion:    blockIngest,
+		ChainBootstrapped: h.bootstrapStatus,
+	}, nil
+}
+
+func (h *Health) NewBestBlock(ID thor.Bytes32) {
```

Comment:
This can be replaced by `*repository.NewTicker()`.

Reply:
Not sure if I'm following, do you mean to replace the `health. NewBestBlock` by a ticker that checks if new blocks have arrived ?
Changing from `node.health.NewBestBlock ` to `health.node.CheckBlockTimestamp` every x seconds ?